### PR TITLE
Define unmarked write-in area in election definition

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Parameterizes unmarked write-in detection by adding a `writeInArea` field to grid positions in the election definition. The `writeInArea` specifies a blank box for each write-in contest option on the ballot where we expect voters to write in a write-in candidate name.

The `writeInArea` is set in a few different places:
- converter-nh-accuvote (for NH elections) - This is based on the previous hard-coded approach for past NH state ballots. It may require tweaking for subsequent elections.
- hmpb/layout (for VX HMPBs) - This box is designed to be above the write-in line and not intersect with the "write-in" label. It varies based on ballot density and column width.

_Review by commit_

## Demo Video or Screenshot
**NH Ballots**
Screenshot 2023-11-08 at 10 13 35 AM Screenshot 2023-11-08 at 10 13 43 AM 

**VX Ballots**
| Density 0 | Density 1 | Density 2 |
|--------|--------|--------|
| Screenshot 2023-11-07 at 4 59 32 PM | Screenshot 2023-11-07 at 5 00 30 PM | Screenshot 2023-11-07 at 4 59 21 PM |

## Testing Plan
- Added write-ins (marked and unmarked) to VX ballot fixtures, added corresponding interpretation tests
- Ensured that existing NH tests (including m17-backup) still yielded the same results for unmarked write-ins. The nice thing about the m17-backup is that our UWI detection gave us the logical results we wanted in every case, so it's a great reference.

I also tried out manually tweaking the m17 election to have bespoke writeInAreas for each contest option (since the contest label in the write-in area is not positioned consistently). This also worked well and yielded the same results.

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Review by eventualbuddha — COMMENTED** [review]
Comments from when it was a draft.

File: libs/ballot-interpreter/test/fixtures/m17-backup/election.json [hunk 1]
```diff
@@ -323,10 +323,7 @@
   ],
   "gridLayouts": [
     {
-      "precinctId": "town-id-00701-precinct-id-",
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter/test/fixtures/m17-backup/election.json:326]: Yes, I think when I made that change I was trying to change as little as possible about the fixture election (partially to test backwards compatibility in our election parsing, partially to preserve the real-world fixture). But I don't care about either of those things as much now

File: libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs [hunk 2]
```diff
@@ -139,18 +139,55 @@ impl TimingMarkGrid {
...
+        let left_after = self
+            .complete_timing_marks
+            .left_rects
+            .get(row_after as usize)?;
+        let right_after = self
+            .complete_timing_marks
+            .right_rects
+            .get(row_after as usize)?;
+        let left = Rect::new(
+            left_before.left(),
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs:171]: Hmm, interesting question. I honestly didn't think about it. I was prototyping and forgot to actually revisit and clean up this code after it seemed to work well.

The timing marks are quite close together so it does seem fairly safe. But it also wouldn't be too hard to account for. What do you think?

File: libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs [hunk 3]
```diff
@@ -161,8 +198,7 @@ impl TimingMarkGrid {
             Rect::new(right.left(), right.top(), timing_mark_width, right.height());
 
         let horizontal_segment = Segment::new(corrected_left.center(), corrected_right.center());
-        let distance_percentage =
-            column as SubPixelUnit / (self.geometry.grid_size.width - 1) as SubPixelUnit;
+        let distance_percentage = column / (self.geometry.grid_size.width - 1) as SubGridUnit;
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs:201]: Yes, good call

File: libs/ballot-interpreter/src/hmpb-rust/interpret.rs [hunk 4]
```diff
@@ -515,26 +515,10 @@ pub fn interpret_ballot_card(
...
-                        image,
-                        grid_layout,
-                        contest_layouts,
-                        scored_bubble_marks,
-                        debug,
-                    )
+                (&front_image, &front_grid, BallotSide::Front, &front_debug),
+                (&back_image, &back_grid, BallotSide::Back, &back_debug),
+                |(image, grid, side, debug)| {
+                    score_write_in_areas(image, grid, grid_layout, sheet_number, side, debug)
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter/src/hmpb-rust/interpret.rs:521]: Yup, you're right.
```
$ cargo --version
cargo 1.69.0 (6e9a83356 2023-04-12)
```
Upgraded, addressed your other comments, and reformatted here: 4223b6f28. Though this particular bit of code still formats the same for me...

**Review by adghayes — COMMENTED** [review]
I won't approve because I'm still unfamiliar with the rust but everything else LGTM.

File: libs/ballot-interpreter/src/hmpb-ts/__snapshots__/interpret.test.ts.snap
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter/src/hmpb-ts/__snapshots__/interpret.test.ts.snap:1]: Ooh good to know, thanks!

**Comment by jonahkagan** [comment]
@eventualbuddha could you approve for the interpreter changes? Or let me know if there's more you think should be addressed (e.g. https://github.com/votingworks/vxsuite/pull/4179#discussion_r1387061782)

**Review by adghayes — APPROVED** [review]
LGTM, although I think we should bump default write-in area threshold to 0.05.

**Pull request merged**